### ISSUE TITLE
Compile issues when using 3.6.0.CR1

### ISSUE THREAD

**Issue by rdruss** [openshiftio-vertx-boosters/vertx-cache-booster#33]

**Comment by cescoffier** [comment]
It's a bit more complicated because it depends on what do we do with the result.

If we expect a `Completable`, using `ignoreElement()` is fine
If we expect a `Single`, using `toSingle` is fine. However, in case of absence, it propagates an error (which is fine in our case).

The resulting class is:

```
public class Cache<K, V> {

    private final Vertx vertx;
    private final RemoteCache<K, V> cache;

    public static <K, V> Single<Cache<K, V>> create(Vertx vertx) {
        Configuration config = new ConfigurationBuilder().addServer().host("cache-server").port(11222).build();
        return vertx.
            <RemoteCache<K, V>>rxExecuteBlocking(
                future -> {
                    RemoteCache<K, V> cache = new RemoteCacheManager(config).getCache();
                    future.complete(cache);
                }
            )
            .toSingle()
            .map(rc -> new Cache<>(vertx, rc));
    }

    private Cache(Vertx vertx, RemoteCache<K, V> rc) {
        this.vertx = vertx;
        this.cache = rc;
    }

    public Completable remove(K key) {
        return vertx.rxExecuteBlocking(
            future -> {
                cache.remove(key);
                future.complete();
            }
        ).ignoreElement();
    }

    public Single<Optional<V>> get(K key) {
        // While this method can use the Maybe type, I found easier to use an Optional.
        return vertx.<Optional<V>>rxExecuteBlocking(future -> {
            V value = cache.get(key);
            future.complete(Optional.ofNullable(value));
        }).toSingle();
    }

    public Completable put(K key, V value, long ttl) {
        return vertx.rxExecuteBlocking(future -> {
            cache.put(key, value, ttl, TimeUnit.SECONDS);
            future.complete();
        }).ignoreElement();
    }

}
```

**Comment by cescoffier** [comment]
I've pushed the fix in the `update-to-vertx-3.6.0` branch.
